Applications/spire/source/charting/cached_chart_model.cpp

```diff
@@ -36,90 +37,135 @@ QtPromise<std::vector<Candlestick>> CachedChartModel::load(ChartValue first,
       });
     }
   }
-  return load_data({first, last});
+  return load_data(get_gaps(first, last), first, last);
 }
 
 connection CachedChartModel::connect_candlestick_slot(
     const CandlestickSignal::slot_type& slot) const {
   return m_chart_model->connect_candlestick_slot(slot);
 }
 
+std::vector<ChartRange> CachedChartModel::get_gaps(ChartValue first,
+    ChartValue last) {
+  if(m_ranges.empty()) {
+    return {{ChartRange{first, last}}};
+  }
+  auto range_points = std::vector<ChartValue>({first, last});
```

Comment:
What is the purpose of the ```vector```? A ```vector``` is used for a dynamically resizable list of elements, but ```range_points``` only ever contains two elements. Use two well named variables instead of a ```vector```.